### PR TITLE
Api server helm chart

### LINKED ISSUE #24: Instantiate API server from Helm Chart
Add a k8s deployment to the helm chart for instantiating the API Server.

### PR DESCRIPTION
**Pull request by maruiz93**
closes #24

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,6 +155,9 @@ Here is an example of the tweaks needed to add `Pods` and `ConfigMaps` to the wa
    ```bash
    helm upgrade -n default --reuse-values -f charts/kubearchive/values.yaml [deployment name] charts/kubearchive --version 0.0.1
    ```
+   > **_NOTE:_**  A tweak is needed to this command for successfully deploying the api-server.
+   Check the section below.
```
Comment (skoved) [inline README.md:159]: I think you should reference the name of the section you are referring to since it's not immediately after this

**Review by skoved — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -163,3 +166,96 @@ Here is an example of the tweaks needed to add `Pods` and `ConfigMaps` to the wa
...
+> [deployment name] charts/kubearchive --version 0.0.1
+> ```
+> This is needed if changes are made to the code.
+
+To test the API expose the port:
+
+```bash
+kubectl port-forward \
+$(kubectl get pods --no-headers -o custom-columns=":metadata.name" | grep api-server) \
+8081:8081
```
Comment (rh-hemartin) [inline README.md:212]: We should expose the service instead:

```suggestion
kubectl -n kubearchive port-forward svc/kubearchive-api-server 8081:8081
```

File: README.md [hunk 3]
```diff
@@ -163,3 +166,96 @@ Here is an example of the tweaks needed to add `Pods` and `ConfigMaps` to the wa
...
+```
+
+And do a query:
+
+```bash
+curl localhost:8081/apis/apps/v1/deployments
+```
+
```
Comment (rh-hemartin) [inline README.md:220]:
```suggestion
To check the logs run:

`kubectl -n kubearchive logs -l app=kubearchive-api-server -f`
```

File: charts/kubearchive/templates/api_server.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# yamllint disable-file
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubearchive-api-server
+  namespace: kubearchive
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/api_server.yaml:7]:
```suggestion
  namespace: {{ .Values.kubearchive.namespace }}
```

File: charts/kubearchive/templates/api_server.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# yamllint disable-file
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubearchive-api-server
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/api_server.yaml:6]:
```suggestion
  name: {{ .Values.apiServer.name }}
```

File: charts/kubearchive/templates/api_server.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          {{- if .Values.apiServer.debug }}
+          command: ["./go/bin/dlv"]
+          args: ["--listen=:40000", "--headless=true", "--api-version=2", "--log", "exec", "/ko-app/api"]
+          {{- end}}
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: {{ .Values.apiServer.name }}
+  namespace: kubearchive
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/api_server.yaml:30]:
```suggestion
  namespace: {{ .Values.kubearchive.namespace }}
```

File: charts/kubearchive/templates/role.yaml [hunk 7]
```diff
@@ -23,3 +23,15 @@ rules:
     {{- end }}
   {{- end }}
 {{- end }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Values.apiServer.clusterRole }}
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/role.yaml:30]: Call everything the same, so its easier tracking it.

```suggestion
  name: {{ .Values.apiServer.name }}
```

File: charts/kubearchive/templates/role_binding.yaml [hunk 8]
```diff
@@ -15,3 +15,16 @@ subjects:
     name: {{ $.Values.kubearchive.serviceAccount }}
     namespace: {{ .name }}
 {{- end }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Values.apiServer.clusterRoleBinding }}
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/role_binding.yaml:22]: Call everything the same so its easier to track.

```suggestion
   name: {{ .Values.apiServer.name }}
```

File: charts/kubearchive/templates/role_binding.yaml [hunk 9]
```diff
@@ -15,3 +15,16 @@ subjects:
...
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Values.apiServer.serviceAccount }}
+    namespace: {{ .Values.kubearchive.namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ .Values.apiServer.clusterRole }}
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/role_binding.yaml:30]:
```suggestion
  name: {{ .Values.apiServer.name }}
```

File: charts/kubearchive/templates/service_account.yaml [hunk 10]
```diff
@@ -13,3 +13,9 @@ metadata:
   name: {{ $.Values.kubearchive.serviceAccount }}
   namespace: {{ .name }}
 {{- end }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.apiServer.serviceAccount }}
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/service_account.yaml:20]:
```suggestion
  name: {{ .Values.apiServer.name }}
```

File: charts/kubearchive/values.yaml [hunk 11]
```diff
@@ -47,6 +47,21 @@ kubearchive:
...
+# values used to create the api-server
+apiServer:
+  name: "kubearchive-api-server"
+  # NOTE - Helm does not resolve ko path. Needs to be set on `helm install` command, e.g:
+  # helm install -n default <chart-name> charts/kubearchive \
+  # --set-string apiServer.image=$(ko build github.com/kubearchive/kubearchive/cmd/api)
+  image: ko://github.com/kubearchive/kubearchive/cmd/api
+  serviceAccount: api
+  clusterRole: api
+  clusterRoleBinding: api
```
Comment (rh-hemartin) [inline charts/kubearchive/values.yaml:59]: If you call everything using `.name`, this can be removed.

File: README.md [hunk 12]
```diff
@@ -163,3 +166,96 @@ Here is an example of the tweaks needed to add `Pods` and `ConfigMaps` to the wa
...
+#### Debug api-server
```
Comment (rh-hemartin) [inline README.md:221]:
```suggestion
#### Remote debugging the API Server from your IDE
```

File: README.md (hunk 3 above)
Comment (maruiz93) [reply README.md:221]: Addressed

File: README.md (hunk 1 above)
Comment (maruiz93) [reply README.md]: Good point! Solved

File: README.md (hunk 2 above)
Comment (maruiz93) [reply README.md]: Makes sense!

File: charts/kubearchive/values.yaml (hunk 11 above)
Comment (maruiz93) [reply charts/kubearchive/values.yaml]: Addressed

**Review by rh-hemartin — APPROVED** [review]

**Review by skoved — APPROVED** [review]

**Pull request merged**